presto-main/src/main/java/com/facebook/presto/type/Re2JRegexp.java

```diff
@@ -88,6 +91,45 @@ public Slice replace(Slice source, Slice replacement)
         }
     }
 
+    public Slice replace(Slice source, UnaryFunctionInterface replaceFunction)
+    {
+        Matcher matcher = re2jPattern.matcher(source);
+        try {
+            matcher.reset();
+            int appendPos = 0;
+            SliceOutput sliceOutput = new DynamicSliceOutput(source.length());
+            while (matcher.find()) {
```

Comment:
According to doc:
`The search begins at the end of the last match, or else the beginning`
Is it OK?

Reply:
Sorry I don't quite get it. Could you be more specific? The doc says the search will start at the beginning of the input when it's called the first time, or when there was no match found when called last time. But for the latter case we won't be calling it again because the while loop will end if it returns false.

Reply:
I was curious if we wanted to support overlapping matches, but it doesn't make sense. Ignore my previous comment.